Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs

```diff
@@ -93,13 +94,22 @@ private void LoadUIBackgroundTexture(ref PBUiBackground sdkModel, ref UIBackgrou
             if (uiBackgroundComponent.Status != LifeCycle.LoadingInProgress ||
                 uiBackgroundComponent.TexturePromise == null ||
                 !frameTimeBudgetProvider.TrySpendBudget() ||
-                !memoryBudgetProvider.TrySpendBudget() ||
-                !uiBackgroundComponent.TexturePromise.Value.TryConsume(World, out StreamableLoadingResult<Texture2D> promiseResult))
+                !memoryBudgetProvider.TrySpendBudget())
                 return;
 
-            // Backgrounds with texture
-            UiElementUtils.SetupDCLImage(ref uiBackgroundComponent.Image, ref sdkModel, promiseResult.Asset);
-            uiBackgroundComponent.Status = LifeCycle.LoadingFinished;
+            try
+            {
+                if (uiBackgroundComponent.TexturePromise.Value.TryConsume(World, out StreamableLoadingResult<Texture2D> promiseResult))
+                {
+                    // Backgrounds with texture
+                    UiElementUtils.SetupDCLImage(ref uiBackgroundComponent.Image, ref sdkModel, promiseResult.Asset);
+                    uiBackgroundComponent.Status = LifeCycle.LoadingFinished;
+                }
+            }
+            catch (Exception e)
```

Comment:
Why do you catch an exception here?
I think (unless you expect `SetupDCLImage` to throw) you should be checking `promiseResult.Succeeded` and if it's not then (`promiseResult.Asset` is null) log an error

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/303/commits/687b1a9f58f5ecd9c26a0b5d89854c69e072069d